conn.go

```diff
@@ -254,14 +253,9 @@ func (s *Session) dialWithoutObserver(ctx context.Context, host *HostInfo, cfg *
 		// be modified after being used.
 		tlsConfig := cfg.tlsConfig
 		if !tlsConfig.InsecureSkipVerify && tlsConfig.ServerName == "" {
-			colonPos := strings.LastIndex(addr, ":")
-			if colonPos == -1 {
```

Comment:
This check might not have been necessary because `addr := host.HostnameAndPort()` would have always added the `:`.